lib/spack/spack/fetch_strategy.py

```diff
@@ -829,9 +829,15 @@ def fetch(self):
             args.append('at revision %s' % self.revision)
         tty.msg("Trying to clone Mercurial repository:", self.url, *args)
 
-        args = ['clone', self.url]
+        args = ['clone']
+
+        if spack.insecure:
+            args.append('--insecure')
```

Comment:
Which unit test runs `hg`? I would like to add `spack.insecure` to cover this line.